parquet-column/src/main/java/org/apache/parquet/column/statistics/Statistics.java

```diff
@@ -191,6 +191,14 @@ public void mergeStatistics(Statistics stats) {
   abstract public byte[] getMinBytes();
 
   /**
+   * Abstract method to return whether the min and max values fit in the given
+   * size.
+   * @param size a size in bytes
+   * @return true iff the min and max values are less than size bytes
+   */
+  abstract public boolean isSmallerThan(long size);
```

Comment:
It's been a while, but I think my logic was this:
* `getMinBytes` and `getMaxBytes` allocate byte arrays and generally too much work for this check. We could cache the bytes, but that requires more work.
* `size` isn't determined by `Statistics` because the metadata converter is responsible for actually converting stats into bytes. This only assumes that the entire min and max bytes will be written.

So we delegate to a stats object to avoid work, but can't delegate to stats because it doesn't do serialization. I could change it to `getSizeHint`, but I don't think that's much better and I'm inclined to leave it as is.